packages/window/src/theme.rs

```diff
@@ -0,0 +1,277 @@
+//!
+//! # Examples
+//! An example of using the theme to determine which class to use.
+//! ```rust
+//! use dioxus::prelude::*;
+//! use dioxus_window::theme::{use_theme, Theme};
+//!
+//! #[component]
+//! fn App() -> Element {
+//!     let theme = use_theme();
+//!     
+//!     // Default to a light theme in the event of an error.
+//!     let class = match theme().unwrap_or(Theme::Light) {
+//!         Theme::Light => "bg-light",
+//!         Theme::Dark => "bg-dark",
+//!     };
+//!
+//!     rsx! {
+//!         div {
+//!             class: "{class}",
+//!             "the current theme is: {theme().unwrap_or(Theme::Light)}"
+//!         }
+//!     }
+//! }
+//! ```
+use dioxus::prelude::*;
+use std::{error::Error, fmt::Display};
+
+/// A color theme.
+///
+/// For any themes other than `light` and `dark`, a [`ThemeError::UnknownTheme`] will be returned.
+/// We may be able to support custom themes in the future.
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+pub enum Theme {
+    /// A light theme, better in direct sunlight.
+    #[default]
+    Light,
+    /// A dark theme, better for the night owls.
+    Dark,
+}
+
+impl Display for Theme {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::Light => write!(f, "light"),
+            Self::Dark => write!(f, "dark"),
+        }
+    }
+}
+
+/// Possible theme errors.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ThemeError {
+    /// Theme is not supported on this platform.
+    Unsupported,
+    /// Failed to get the system theme.
+    CheckFailed,
+    /// System returned an unknown theme.
+    UnknownTheme,
+}
+
+impl Error for ThemeError {}
+impl Display for ThemeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::Unsupported => write!(f, "the current platform is not supported"),
+            Self::CheckFailed => write!(
+                f,
+                "the system returned an error while checking the color theme"
+            ),
+            Self::UnknownTheme => write!(
+                f,
+                "the system provided a theme other than `light` or `dark`"
+            ),
+        }
+    }
+}
+
+type ThemeResult = Result<Theme, ThemeError>;
+
+/// Get a signal to the system theme.
+///
+/// On first run, the result will be [`ThemeError::Unsupported`]. This is to prevent hydration from failing.
+/// After the client runs, the theme will be tracked and updated with accurate values.
+///
+/// # Examples
+///
+/// ```rust
+/// use dioxus::prelude::*;
+/// use dioxus_window::theme::{use_theme, Theme};
+///
+/// #[component]
+/// fn App() -> Element {
+///     let theme = use_theme();
+///
+///     rsx! {
+///         p {
+///             "the current theme is: {theme().unwrap_or(Theme::Light)}"
+///         }
+///     }
+/// }
+/// ```
+pub fn use_theme() -> ReadOnlySignal<ThemeResult> {
```

Comment:
Maybe its just me but I think something like `use_system_theme` was better. `use_theme` is very generic imo. `use_preferred_theme` is a good candidate too.